scripts/gradle/gradle-check.sh

```diff
@@ -43,14 +44,31 @@ if [ -z "$QUEUE_URL" ] || [ "$QUEUE_URL" != "null" ]; then
     if [ -z "$WORKFLOW_URL" ] || [ "$WORKFLOW_URL" != "null" ]; then
 
         RUNNING="true"
+        RETRY_COUNT=0
 
         echo "Waiting for Jenkins to complete the run"
         while [ "$RUNNING" = "true" ] && [ "$TIMEPASS" -le "$TIMEOUT" ]; do
             echo "Still running, wait for another 30 seconds before checking again, max timeout $TIMEOUT"
             echo "Jenkins Workflow Url: $WORKFLOW_URL"
             TIMEPASS=$(( TIMEPASS + 30 )) && echo time pass: $TIMEPASS
             sleep 30
-            RUNNING=$(curl -s -XGET ${WORKFLOW_URL}api/json | jq --raw-output .building)
+
+            CURL_RESPONSE_CODE=$(curl -s -o /dev/null -w "%{http_code}" -XGET "${WORKFLOW_URL}api/json")
```

Comment:
Are we sure 200 is the only response code returned. Why not leave the previous code as is and add retry if exit code is 1?

Reply:
Thanks for the comment, @rishabh6788 , let me do retry on non-0 exit code